src/utils/index.tsx

```diff
@@ -0,0 +1,167 @@
+		const reader = new FileReader();
+		reader.readAsDataURL(file);
+		reader.onload = () => resolve(reader.result);
+		reader.onerror = error => reject(error);
+	});
+
+export const getImageFile = async (base64Data: any, projectName: any) => {
+	const imageFile = await fetch(base64Data)
+		.then(res => res.blob())
+		.then(blob => {
+			return new File([blob], projectName);
+		});
+	return imageFile;
+};
+
+// export async function getEtherscanTxs(address, apolloClient = false, isDonor = false) {
+//   try {
+//     const apiKey = process.env.NEXT_PUBLIC_ETHERSCAN_API_KEY
+//     const api = process.env.NEXT_PUBLIC_NETWORK_ID === '3' ? 'api-ropsten' : 'api'
+//     const balance = await fetch(
+//       `https://${api}.etherscan.io/api?module=account&action=balance&address=${address}&tag=latest&apikey=${apiKey}`
+//     )
+//       .then(response => response.json())
+//       .then(data => {
+//         return data?.result
+//       })
+
+//     return await fetch(
+//       `https://${api}.etherscan.io/api?module=account&action=txlist&address=${address}&startblock=0&endblock=99999999&sort=asc&apikey=${apiKey}`
+//     )
+//       .then(response => response.json())
+//       .then(async data => {
+//         const modified = []
+//         if (data?.status === '0' || typeof data?.result === 'string') {
+//           return {
+//             balance,
+//             txs: [],
+//             error: data?.result
+//           }
+//         }
+//         for (const t of data?.result) {
+//           const extra = apolloClient
+//             ? await apolloClient?.query({
+//                 query: isDonor ? GET_USER_BY_ADDRESS : GET_PROJECT_BY_ADDRESS,
+//                 constiables: {
+//                   address: isDonor
+//                     ? Web3.utils.toChecksumAddress(t?.from)
+//                     : Web3.utils.toChecksumAddress(t?.to)
+//                 }
+//               })
+//             : null
+//           modified.push({
+//             ...t,
+//             extra: extra?.data || null,
+//             donor: t.from,
+//             createdAt: t.timeStamp,
+//             currency: 'ETH'
+//           })
+//         }
+//         return {
+//           balance,
+//           txs: modified
+//         }
+//       })
+//   } catch (error) {
+//     console.log({ error })
+//   }
+// }
+
+export const getERC20List = ERC20List;
+
+export async function checkIfURLisValid(checkUrl: string) {
+	let url = checkUrl;
+	if (!/^(?:f|ht)tps?:\/\//.test(checkUrl)) {
+		url = 'https://' + url;
+	}
+	const pattern = new RegExp(
+		'^(https?:\\/\\/)?' + // protocol
+			'((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|' + // domain name
+			'((\\d{1,3}\\.){3}\\d{1,3}))' + // OR ip (v4) address
+			'(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' + // port and path
+			'(\\?[;&a-z\\d%_.~+=-]*)?' + // query string
+			'(\\#[-a-z\\d_]*)?$',
+		'i',
+	);
+	return !!pattern.test(url);
+}
+
+export const switchNetwork = (currentNetworkId?: any) => {
```

Comment:
We have a function for changing networks wich supports MetaMask and Torus, please use that and delete this one